Added search progress spinner in search bar

DeepaliNemade839 opened the pull request:
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

Added a search progress spinner in search bar with spinner start+visible and stop+hide control given to user.

SearchBar.swift:
- inside search bar, added a new element for progress spinner which is enclosing the clear button in search bar.
- User will have control over hide/show and start/stop animating the progress spinner.
- tint color of progress spinner is kept same as that of other elements on search bar

NavigationControllerDemoController.swift
- Made the root controller to implement the search bar delegate
- Added a switch in root controller's view to let user select if they want to show progress spinner when user hits search or not
- upon updating the text in search bar or cancel button press, stopping+hiding the progress spinner 
- when user hits search for a keyword, the progress spinner appears and starts spinning (if above switch is enabled)

### Verification

- Validated that user is able to show and hide the spinner and start and stop the spinner animation depending on switch status
- validated that the other search bar controls are working as before.

| Before                                                                  | After                                                                 |
|----------------------------------------------|--------------------------------------------|
| No progress Spinner was present in search bar | Progress spinner ui element is added inside search bar surrounding clear button |

### Screenshots

![Simulator Screen Shot - iPhone 8 Plus - 2020-08-15 at 00 08 10](https://user-images.githubusercontent.com/68395453/90282224-9a5fa000-de8b-11ea-9a5e-de698b402694.png)

![Simulator Screen Shot - iPhone 8 Plus - 2020-08-15 at 00 27 05](https://user-images.githubusercontent.com/68395453/90283706-3b4f5a80-de8e-11ea-8eb0-da4927eaf1c8.png)

![Simulator Screen Shot - iPhone 8 Plus - 2020-08-15 at 00 26 54](https://user-images.githubusercontent.com/68395453/90283715-40140e80-de8e-11ea-8f4e-89ad059ad517.png)

![Simulator Screen Shot - iPhone 8 Plus - 2020-08-15 at 00 26 44](https://user-images.githubusercontent.com/68395453/90283718-40aca500-de8e-11ea-83cf-159552f0a03f.png)

![Simulator Screen Shot - iPhone 8 Plus - 2020-08-15 at 00 28 52](https://user-images.githubusercontent.com/68395453/90283746-50c48480-de8e-11ea-97b6-b869a95c96f1.png)

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [x] VoiceOver and Keyboard Accessibility
- [x] Internationalization and Right to Left layouts
- [x] Different resolutions (1x, 2x, 3x)
- [x] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)


###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/fluentui-apple/pull/174)

harrieshin commented:
in your screenshot the spinner colors stays the same which is not true in toolkit design.

harrieshin commented on ios/FluentUI/Controls/SearchBar.swift:
```diff
@@ -214,6 +225,13 @@ open class SearchBar: UIView {
         return clearButton
     }()
 
+    //indicates search in progress
```
needs to be apple documentation commenting style for clarity

DeepaliNemade839 replied:
fixed

---
In reply to: [465866211](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r465866211) [](ancestors = 465866211)

DeepaliNemade839 replied:
Thanks :)

---
In reply to: [465912220](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r465912220) [](ancestors = 465912220)

harrieshin commented on ios/FluentUI/Controls/SearchBar.swift:
```diff
@@ -330,6 +348,16 @@ open class SearchBar: UIView {
+        progressSpinner.translatesAutoresizingMaskIntoConstraints = false
+
+        constraints.append(progressSpinner.leadingAnchor.constraint(equalTo: searchTextField.trailingAnchor, constant: Constants.clearButtonLeadingInset))
+        constraints.append(progressSpinner.widthAnchor.constraint(equalToConstant: Constants.clearButtonWidth))
```
if there are leading and trailing anchors is there a need for width anchor?

DeepaliNemade839 replied:
copied the constraints from below and missed evaluating them. Will take a look and remove redundant.

---
In reply to: [465867515](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r465867515) [](ancestors = 465867515)

DeepaliNemade839 replied:
removed redundant  in iteration 2

---
In reply to: [465882946](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r465882946) [](ancestors = 465882946,465867515)

harrieshin commented on ios/FluentUI.Demo/FluentUI.Demo/Demos/NavigationControllerDemoController.swift:
```diff
@@ -150,7 +151,27 @@ class NavigationControllerDemoController: DemoController {
 // MARK: - NavigationControllerDemoController: UIGestureRecognizerDelegate
 
-extension NavigationControllerDemoController: UIGestureRecognizerDelegate {
+extension NavigationControllerDemoController: UIGestureRecognizerDelegate, SearchBarDelegate {
```
can we make sure there are ways for this demo to test non spinner case?

DeepaliNemade839 replied:
as implemented in delegate, when user press search button, i.e. on searchBarDidRequestSearch invocation we are showing the animated progress bar.

---
In reply to: [465868161](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r465868161) [](ancestors = 465868161)

rdeassis replied:
As I mentioned before, I think it'd be nice to have a toggle that you can set the spinner visibility.
Right now it's hard to discover the spinner since we have to type text and hit "Search" to see it.

DeepaliNemade839 replied:
I find following issues in adding a switch on the nav bar buttons:
1. it doesn't get sized/positioned correctly as it tried implementing it. Also, it gets primary color which gets mixed up with the nav bar's color
2. when we tap on search button, search bar moves up, hiding this switch and hence at that time we don't have a switch.
Also, I think it is far from a real use when users will want a spinner.
Couple of options that I could think of other than the approach in CR are:
1. having a separate demo view controller similar which showcases search bar's. But then the only extra functionality that we are displaying there would be the spinner
2. Instead of putting the switch in navigation bar button items, we can add it as subview [label + switch]. But the addRow functionality is available in DemoController which isn't inherited by the controllers that we are showcasing here inside navigation controller's demo.
I would like to discuss it before making change.

---
In reply to: [468751778](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r468751778) [](ancestors = 468751778)

harrieshin commented on ios/FluentUI/Controls/SearchBar.swift:
```diff
@@ -23,6 +24,7 @@ public extension Colors {
             public static var cancelButton: UIColor = LightContent.text
             public static var clearIcon = UIColor(light: iconOnAccent, dark: textSecondary)
             public static var placeholderText = UIColor(light: textOnAccent, dark: textSecondary)
+            public static var progressSpinner: UIColor = placeholderText
```
seems incorrect to me. we should show primaryTint10 color for light mode if background is app primary color (like communicationblue)

DeepaliNemade839 replied:
updated the color. But i could not identity the difference in color for light mode on primary background in demo app.

---
In reply to: [465968168](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r465968168) [](ancestors = 465968168)

harrieshin commented on ios/FluentUI/Controls/SearchBar.swift:
```diff
@@ -178,21 +178,21 @@ open class SearchBar: UIView {
     weak var navigationController: NavigationController?
 
-    //used to hide the cancelButton in non-active states
+    /// used to hide the cancelButton in non-active states
```
for private function likes these // is the right format. /// is used for public classes that you are showing documentation when developer right clicks on the class to find out more info.

DeepaliNemade839 replied:
fixed in 3rd iteration

---
In reply to: [467169551](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r467169551) [](ancestors = 467169551)

DeepaliNemade839 commented on ios/FluentUI/Controls/SearchBar.swift:
```diff
@@ -304,7 +322,7 @@ open class SearchBar: UIView {
         constraints.append(searchTextFieldBackgroundViewTrailing)
 
-        //search icon container
+        /// search icon container
```
>/// [](start = 8, length = 3)

revert

DeepaliNemade839 replied:
done

---
In reply to: [467852130](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r467852130) [](ancestors = 467852130)

Victa commented:
> in your screenshot the spinner colors stays the same which is not true in toolkit design.

- Light mode / primary (= blue nav bar): `Accent color Tint 10`
- Light mode / whiter: `Gray 300`
- Dark mode / default: `Gray 100`

rdeassis commented on ios/FluentUI/Controls/SearchBar.swift:
```diff
@@ -14,6 +14,7 @@ public extension Colors {
             public static var cancelButton = UIColor(light: textSecondary, dark: LightContent.cancelButton)
             public static var clearIcon = UIColor(light: iconPrimary, dark: LightContent.clearIcon)
             public static var placeholderText = UIColor(light: textSecondary, dark: LightContent.placeholderText)
+            public static var progressSpinner: UIColor = DarkContent.searchIcon
```
@Victa already mentioned the colors are incorrect:

Light mode / primary (= blue nav bar): Accent color Tint 10
Light mode / white: Gray 300
Dark mode / default: Gray 100

This is being used on the Light mode/ white, right? It should be gray300 instead of the current gray500.

DarkContent {
public static var **searchIcon** = UIColor(light: **iconPrimary**, dark: LightContent.searchIcon)
}

@objc public static let **iconPrimary** = UIColor(light: **gray500**, lightHighContrast: gray700, dark: .white)

rdeassis replied:
Also, could you update the screenshots once you fix the spinner colors? Thank you! :)

DeepaliNemade839 replied:
updated colors for spinner.

---
In reply to: [468189125](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r468189125) [](ancestors = 468189125)

DeepaliNemade839 replied:
Updated screenshots.

---
In reply to: [468196497](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r468196497) [](ancestors = 468196497)

rdeassis commented on ios/FluentUI/Controls/SearchBar.swift:
```diff
@@ -119,6 +120,15 @@ open class SearchBar: UIView {
+        func progressSpinnerColor(for window: UIWindow) -> UIColor {
+            switch self {
+            case .lightContent:
+                return UIColor(light: Colors.primaryTint10(for: window), dark: Colors.iconSecondary)
```
Same here about the colors:

It's using gray500 instead of gray100 defined by @Victa 

Light mode / primary (= blue nav bar): Accent color Tint 10
Light mode / white: Gray 300
Dark mode / default: **Gray 100**

@objc public static let iconSecondary = UIColor(light: gray400, lightHighContrast: gray600, dark: **gray500**, darkHighContrast: gray300, darkElevated: gray400)

DeepaliNemade839 replied:
updated colors. 

---
In reply to: [468190424](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r468190424) [](ancestors = 468190424)

rdeassis commented on ios/FluentUI.Demo/FluentUI.Demo/Demos/NavigationControllerDemoController.swift:
```diff
@@ -150,7 +151,27 @@ class NavigationControllerDemoController: DemoController {
+
+       func searchBarDidBeginEditing(_ searchBar: SearchBar) {
+               searchBar.progressSpinner.stopAnimating()
+               searchBar.progressSpinner.isHidden = true
```
Or is this the property to make the spinner optional as @Victa mentioned [here](https://ghe-us.microsoft.com/Microsoft-Design-System-Studio/Design-Toolkit-iOS/issues/1#issuecomment-332749)?
("I suggest to make it optional.")

If that's the case, I think there should be a toggle switch to set the spinner on/off instead of demoing in the delegate. Setting these here is redundant with the hidesWhenStopped property of the ActivityIndicatorView.

DeepaliNemade839 replied:
removed. Thanks

---
In reply to: [468181951](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r468181951) [](ancestors = 468181951)

DeepaliNemade839 replied:
On the demo view there is no place to put an extra switch element. Hence, tried to give search start and stop as the progress spinner's start and stop entry point which will mostly be the case in real use case.
Let me know your thoughts.

---
In reply to: [468195652](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r468195652) [](ancestors = 468195652)

rdeassis replied:
Thanks for adding the switch to showcase the optional spinner! 🥇

DeepaliNemade839 replied:
Thanks!

---
In reply to: [470779096](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r470779096) [](ancestors = 470779096)

DeepaliNemade839 commented:
Updated colors as mentioned by Victor and updated screenshots in pull request

---
In reply to: [669304788](https://github.com/microsoft/fluentui-apple/pull/174#issuecomment-669304788) [](ancestors = 669304788)

DeepaliNemade839 commented:
Thanks Victor. Updated colors as mentioned here.

---
In reply to: [671489608](https://github.com/microsoft/fluentui-apple/pull/174#issuecomment-671489608) [](ancestors = 671489608)

DeepaliNemade839 commented on ios/FluentUI/Controls/SearchBar.swift:
```diff
@@ -361,6 +390,9 @@ open class SearchBar: UIView {
         searchTextField.tintColor = style.tintColor
         clearButton.tintColor = style.clearIconColor
+        if let window = window {
+            progressSpinner.tintColor = style.progressSpinnerColor(for: window)
```
Thanks :) I did not notice tint being used :( Updated to set color and now it is working absolutely fine.

---
In reply to: [468178614](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r468178614) [](ancestors = 468178614)

DeepaliNemade839 commented on ios/FluentUI/Controls/SearchBar.swift:
```diff
@@ -214,6 +225,13 @@ open class SearchBar: UIView {
     }()
 
+    //indicates search in progress
+    @objc public var progressSpinner: ActivityIndicatorView = {
```
Done

---
In reply to: [465905184](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r465905184) [](ancestors = 465905184)

DeepaliNemade839 commented on ios/FluentUI/Controls/SearchBar.swift:
```diff
@@ -23,6 +24,7 @@ public extension Colors {
             public static var cancelButton: UIColor = LightContent.text
             public static var clearIcon = UIColor(light: iconOnAccent, dark: textSecondary)
             public static var placeholderText = UIColor(light: textOnAccent, dark: textSecondary)
+            public static var progressSpinner: UIColor = placeholderText
```
as i was using tint color to set color of progress spinner, the color wasn't getting updated. Fixed the color and updated screenshots.

---
In reply to: [465909151](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r465909151) [](ancestors = 465909151)

rdeassis commented on ios/FluentUI/Controls/SearchBar.swift:
```diff
@@ -361,6 +389,9 @@ open class SearchBar: UIView {
         searchTextField.tintColor = style.tintColor
         clearButton.tintColor = style.clearIconColor
+        if let window = window {
+            progressSpinner.color = style.progressSpinnerColor(for: window)
```
You need to add this to the ActivityIndicatorView to make sure that the spinner updates its color if it's being displayed while dark and light modes are switched:

    public override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
        super.traitCollectionDidChange(previousTraitCollection)
        if #available(iOS 13, *) {
            if previousTraitCollection?.userInterfaceStyle != traitCollection.userInterfaceStyle {
                setupLoaderLayer()
            }
        }
    }

harrieshin replied:
really? UIColor(light:dark:) didn't do it for you?

rdeassis replied:
It doesn't because the ActivityIndicatorView sets a CGColor on its layer instead of a UIColor. So, unfortunately we don't get the color update for free. :(

harrieshin replied:
oh yeah great catch!

DeepaliNemade839 replied:
Incorporated the comment and verified that upon switching the mode, the color is getting updated. Thanks

---
In reply to: [470055237](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r470055237) [](ancestors = 470055237)

DeepaliNemade839 replied:
Done

---
In reply to: [468750117](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r468750117) [](ancestors = 468750117)

harrieshin commented on ios/FluentUI.Demo/FluentUI.Demo/Demos/NavigationControllerDemoController.swift:
```diff
@@ -150,7 +151,24 @@ class NavigationControllerDemoController: DemoController {
+       }
+
+       func searchBar(_ searchBar: SearchBar, didUpdateSearchText newSearchText: String?) {
+               // Do Nothing
```
remove?

DeepaliNemade839 replied:
can't be removed. It is not optional.

---
In reply to: [468818602](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r468818602) [](ancestors = 468818602)

harrieshin replied:
i meant the comment

DeepaliNemade839 replied:
removed :)

---
In reply to: [470124917](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r470124917) [](ancestors = 470124917)

harrieshin commented on ios/FluentUI/Controls/SearchBar.swift:
```diff
@@ -229,6 +239,12 @@ open class SearchBar: UIView {
 
     private var originalIsNavigationBarHidden: Bool = false
 
+    ///indicates search in progress
```
nit space

DeepaliNemade839 replied:
Fixed

---
In reply to: [469549952](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r469549952) [](ancestors = 469549952)

harrieshin commented on ios/FluentUI/Controls/ActivityIndicatorView.swift:
```diff
@@ -154,6 +154,15 @@ open class ActivityIndicatorView: UIView {
+        super.traitCollectionDidChange(previousTraitCollection)
+        if #available(iOS 13, *) {
+            if previousTraitCollection?.userInterfaceStyle != traitCollection.userInterfaceStyle {
+            setupLoaderLayer()
```
nit: spacing?

DeepaliNemade839 replied:
Sorry, Swiftlint didn't work on it. Corrected

---
In reply to: [470111446](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r470111446) [](ancestors = 470111446)

rdeassis commented on ios/FluentUI.Demo/FluentUI.Demo/Demos/NavigationControllerDemoController.swift:
```diff
@@ -173,6 +178,43 @@ class RootViewController: UIViewController, UITableViewDataSource, UITableViewDe
+        itemRow.layoutMargins = UIEdgeInsets(top: 10, left: 10, bottom: 10, right: 10)
+        itemRow.translatesAutoresizingMaskIntoConstraints = false
+
+        let label = Label(style: .subhead, colorStyle: .regular)
```
nit: can we name this searchSpinnerSwitchLabel?

DeepaliNemade839 replied:
Comment Incorporated

---
In reply to: [470762551](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r470762551) [](ancestors = 470762551)

rdeassis commented on ios/FluentUI.Demo/FluentUI.Demo/Demos/NavigationControllerDemoController.swift:
```diff
@@ -173,6 +178,43 @@ class RootViewController: UIViewController, UITableViewDataSource, UITableViewDe
+            itemsContainer.topAnchor.constraint(equalTo: itemRow.topAnchor),
+            itemsContainer.bottomAnchor.constraint(equalTo: itemRow.bottomAnchor),
+            itemsContainer.leadingAnchor.constraint(equalTo: itemRow.leadingAnchor),
+            itemsContainer.trailingAnchor.constraint(equalTo: itemRow.trailingAnchor)
```
searchSpinnerSwitchLabel.centerYAnchor.constraint(equalTo: itemsContainer.centerYAnchor),
			searchSpinnerSwitch.centerYAnchor.constraint(equalTo: itemsContainer.centerYAnchor)

Justo to align the label and toggle switch...

DeepaliNemade839 replied:
Added constraints and now they looks aligned.

---
In reply to: [470763039](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r470763039) [](ancestors = 470763039)

rdeassis commented on ios/FluentUI.Demo/FluentUI.Demo/Demos/NavigationControllerDemoController.swift:
```diff
@@ -173,6 +178,43 @@ class RootViewController: UIViewController, UITableViewDataSource, UITableViewDe
+        itemRow.distribution = .equalCentering
+        itemRow.alignment = .leading
+        itemRow.isLayoutMarginsRelativeArrangement = true
+        itemRow.layoutMargins = UIEdgeInsets(top: 10, left: 10, bottom: 10, right: 10)
```
Can you set the left and right insets to 15 to match the icons of the table view cells below?

rdeassis commented on ios/FluentUI/Controls/SearchBar.swift:
```diff
@@ -330,12 +350,20 @@ open class SearchBar: UIView {
         constraints.append(clearButton.leadingAnchor.constraint(equalTo: searchTextField.trailingAnchor, constant: Constants.clearButtonLeadingInset))
-        constraints.append(clearButton.widthAnchor.constraint(equalToConstant: Constants.clearButtonWidth))
         constraints.append(clearButton.heightAnchor.constraint(equalTo: clearButton.widthAnchor))
         constraints.append(clearButton.trailingAnchor.constraint(equalTo: searchTextFieldBackgroundView.trailingAnchor, constant: -1 * Constants.clearButtonTrailingInset))
```
I think this might need to change since the spinner is touching the edge of the search bar when it's visible.

rdeassis replied:
I think setting clearButtonTrailingInset to 10.0 is the ideal change here.

DeepaliNemade839 replied:
Done

---
In reply to: [470774158](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r470774158) [](ancestors = 470774158)

DeepaliNemade839 replied:
Fixed

---
In reply to: [470771214](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r470771214) [](ancestors = 470771214)

rdeassis commented on ios/FluentUI.Demo/FluentUI.Demo/Demos/NavigationControllerDemoController.swift:
```diff
@@ -173,6 +178,43 @@ class RootViewController: UIViewController, UITableViewDataSource, UITableViewDe
+        itemRow.translatesAutoresizingMaskIntoConstraints = false
+
+        let label = Label(style: .subhead, colorStyle: .regular)
+        label.text = "Show search spinner when user hits search"
```
nit: how about changing this string to: "Show spinner while using the search bar"?

DeepaliNemade839 replied:
Comment incorporated

---
In reply to: [470783794](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r470783794) [](ancestors = 470783794)

rdeassis commented on ios/FluentUI.Demo/FluentUI.Demo/Demos/NavigationControllerDemoController.swift:
```diff
@@ -104,6 +105,9 @@ class NavigationControllerDemoController: DemoController {
             content.allowsCellSelection = true
         }
 
+        let searchBarView = accessoryView as! SearchBar
```
This is crashing when you try to show without accessory:

Fatal error: Unexpectedly found nil while unwrapping an Optional value

DeepaliNemade839 replied:
Fixed

---
In reply to: [470790975](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r470790975) [](ancestors = 470790975)

DeepaliNemade839 commented on ios/FluentUI.Demo/FluentUI.Demo/Demos/NavigationControllerDemoController.swift:
```diff
@@ -232,6 +278,7 @@ class RootViewController: UIViewController, UITableViewDataSource, UITableViewDe
 
     override func viewDidLoad() {
         super.viewDidLoad()
+        container.addArrangedSubview(searchProgressSpinnerSwitchView)
```
>container.addArrangedSubview(searchProgressSpinnerSwitchView) [](start = 8, length = 61)

don't show when we do not have accessory view

DeepaliNemade839 replied:
Fixed. Added Screenshots for view without accessory

---
In reply to: [470798570](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r470798570) [](ancestors = 470798570)

DeepaliNemade839 commented on ios/FluentUI/Controls/SearchBar.swift:
```diff
@@ -330,12 +350,20 @@ open class SearchBar: UIView {
         constraints.append(searchTextField.centerYAnchor.constraint(equalTo: searchTextFieldBackgroundView.centerYAnchor))
         constraints.append(searchTextField.heightAnchor.constraint(equalTo: searchTextFieldBackgroundView.heightAnchor, constant: -2 * Constants.searchTextFieldVerticalInset))
 
+        //progressSpinner
```
Fixed

---
In reply to: [470738644](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r470738644) [](ancestors = 470738644)

DeepaliNemade839 commented on ios/FluentUI.Demo/FluentUI.Demo/Demos/NavigationControllerDemoController.swift:
```diff
@@ -340,14 +387,38 @@ class RootViewController: UIViewController, UITableViewDataSource, UITableViewDe
         msfNavigationController?.expandNavigationBar(animated: true)
+        container.insertArrangedSubview(searchProgressSpinnerSwitchView, at: 0 /* index */)
+    }
+
```
removed

---
In reply to: [470741628](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r470741628) [](ancestors = 470741628)

DeepaliNemade839 commented on ios/FluentUI.Demo/FluentUI.Demo/Demos/NavigationControllerDemoController.swift:
```diff
@@ -173,6 +178,43 @@ class RootViewController: UIViewController, UITableViewDataSource, UITableViewDe
+        itemRow.addArrangedSubview(label)
+
+        let searchSpinnerSwitch = UISwitch()
+        searchSpinnerSwitch.isOn = false
```
set to true

---
In reply to: [470712622](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r470712622) [](ancestors = 470712622)

rdeassis commented on ios/FluentUI.Demo/FluentUI.Demo/Demos/NavigationControllerDemoController.swift:
```diff
@@ -173,6 +178,45 @@ class RootViewController: UIViewController, UITableViewDataSource, UITableViewDe
+        itemRow.translatesAutoresizingMaskIntoConstraints = false
+
+        let searchSpinnerSwitchLabel = Label(style: .subhead, colorStyle: .regular)
+        searchSpinnerSwitchLabel.text = "Show spinner while using the search bar:"
```
super nit: I think we don't need the ":" at the end of the string

DeepaliNemade839 replied:
removed

---
In reply to: [470811273](https://github.com/microsoft/fluentui-apple/pull/174#discussion_r470811273) [](ancestors = 470811273)

rdeassis replied:
Thank you! :)
